azurerm/internal/services/securitycenter/server_vulnerability_assessment_resource.go

```diff
@@ -0,0 +1,167 @@
+package securitycenter
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/security/mgmt/v3.0/security"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceServerVulnerabilityAssessment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServerVulnerabilityAssessmentCreate,
+		Read:   resourceServerVulnerabilityAssessmentRead,
+		Delete: resourceServerVulnerabilityAssessmentDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"virtual_machine_id": {
```

Comment:
Seems ServerVulnerabilityAssessment supports servers, which can be of type “Microsoft.Compute/virtualMachines”, But also “[Microsoft.HybridCompute/machines](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/stable/2020-08-02/HybridCompute.json)”.

Reply:
I can not find any information on how to create or work with Microsoft.HybridCompute resources from terraform. @neil-yechenwei do you have any information on how to work with HybridCompute in terraform? 
I will need some help in order to know how to setup something where I could test and verify HybridCompute support.

Reply:
Terraform doesn't support Microsoft.HybridCompute/machines for now. I assume you can create/test this resource with [rest api](https://docs.microsoft.com/en-us/rest/api/hybridcompute/machines/createorupdate). I assume the resource `azurerm_server_vulnerability_assessment` could be made more generic to support both of them if ServerVulnerabilityAssessment API also supports Microsoft.HybridCompute/machines?

Reply:
I have added support for Microsoft.HybridCompute/machines (Azure ARC). But I have kept the name virtual_machine_id for the property since that seems to be most consistent with other resources. Or does anybody have any suggestions on a better name?

Reply:
How about `target_machine_id`?